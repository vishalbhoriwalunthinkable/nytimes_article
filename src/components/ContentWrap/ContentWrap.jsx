import React from "react";
import PropTypes from "prop-types";
import { CircularProgress, makeStyles } from "@material-ui/core";

const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    alignItems: "center",
    width: "100%",
  },
}));

export default function ContentWrap({ isFetching, isError, children, style }) {
  const classes = useStyles();
  if (isError) {
    return <div className="ContentWrap">Something went wrong...</div>;
  }

  if (isFetching) {
    return (
      <div
        className={classes.root}
        style={style}
        data-testid="loading-component"
      >
        <CircularProgress />
      </div>
    );
  }
  return children;
}

ContentWrap.defaultProps = {
  isError: false,
  style: { margin: "0px auto", width: "10%", minHeight: "200px" },
};

ContentWrap.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  children: PropTypes.element.isRequired,
  style: PropTypes.object,
};
