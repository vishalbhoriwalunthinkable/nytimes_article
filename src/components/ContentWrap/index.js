import ContentWrap from "./ContentWrap";

export default ContentWrap;
