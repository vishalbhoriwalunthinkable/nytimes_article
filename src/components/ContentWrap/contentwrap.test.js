import React from "react";
import renderer from "react-test-renderer";

import ContentWrap from "./ContentWrap";

it("ContentWrap isFetching true", () => {
  const tree = renderer
    .create(
      <ContentWrap isFetching={true}>
        <div>Child</div>
      </ContentWrap>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("ContentWrap when isFetching false", () => {
  const tree = renderer
    .create(
      <ContentWrap isFetching={false}>
        <div>Child</div>
      </ContentWrap>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("ContentWrap when isError true", () => {
  const tree = renderer
    .create(
      <ContentWrap isFetching={false} isError={true}>
        <div>Child</div>
      </ContentWrap>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("ContentWrap when isError false", () => {
  const tree = renderer
    .create(
      <ContentWrap isFetching={false} isError={false}>
        <div>Child</div>
      </ContentWrap>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
