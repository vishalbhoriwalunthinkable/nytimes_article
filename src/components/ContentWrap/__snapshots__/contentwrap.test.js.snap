// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentWrap isFetching true 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="loading-component"
  style={
    Object {
      "margin": "0px auto",
      "minHeight": "200px",
      "width": "10%",
    }
  }
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`ContentWrap when isError false 1`] = `
<div>
  Child
</div>
`;

exports[`ContentWrap when isError true 1`] = `
<div
  className="ContentWrap"
>
  Something went wrong...
</div>
`;

exports[`ContentWrap when isFetching false 1`] = `
<div>
  Child
</div>
`;
